### PR TITLE
feat: add useAxios hook and tests

### PR DESCRIPTION
**Pull request by archit7448**
### Issue 
- Write useAxios hook and  tests for it.

### Files  to change
- ```jest.config.js```  for test setup
-  ```package.json```
      - Update react and react-testing-library version 
      - Add msw and axios
      - Inside scripts add test-coverage command.
- ```src/custom-hooks/useFetch.js```  change URL methods of passing change with props.

### Files to add 
- ```setupTests.ts```  add teardown
- ```mocks/server.ts``` add server
- ```mocks/handler.ts``` add handler
-  ```hooks/useAxios.ts``` feat: useFetch hook.
- ```__tests__/useAxios.ts``` test for useFetch hook.
  
 ### UseAxios hook 

 - The `useAxios` hook is a custom React hook for making HTTP requests with the `axios` library. It returns an object with three properties: `data`, `loading`, and `error`. The `data` property contains the response data, the `loading` property is a boolean that indicates whether the request is in progress, and the `error` property contains any error that occurred during the request.
 - The hook accepts a URL, an HTTP method (defaults to `GET`), and a configuration object that can be used to pass additional options to `axios`
 
### UseAxios hook test 

- The tests for the `useAxios` hook use the `msw` library to mock HTTP requests and verify that the hook returns the expected values for different scenarios, such as successful requests and failed requests.
- `get` , `patch` and `reject` request during testing.

### CODE REVIEW

File: src/__tests__/useAxios.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { renderHook, waitFor } from "@testing-library/react";
```
Comment (vvaibhavdesai) [inline src/__tests__/useAxios.ts:1]: can we sort the orders in the format of 
external library imports first 
then local library imports 
then local components and then at the last all the utilities

**Review by vvaibhavdesai — CHANGES_REQUESTED** [review]

File: src/__tests__/useAxios.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { rest } from "msw";
+import { renderHook, waitFor } from "@testing-library/react";
+import { useAxios } from "../hooks/useAxios";
+import { server } from "../mocks/server";
+
+
+
+describe("useAxios ", () => {
+  it("should make an HTTP GET request and return the response data", async () => {
+    const { result } = renderHook(() => useAxios("https://real-dev-squad.com"));
```
Comment (Neha) [inline src/__tests__/useAxios.ts:11]: Move the URL to constant

File: src/__tests__/useAxios.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    expect(result.current.loading).toBe(true);
+    expect(result.current.data).toBe(null);
+    expect(result.current.error).toBe(null);
+
+    await waitFor(() => expect(result.current.loading).toBe(false));
+    await waitFor(() =>
+      expect(result.current.data).toStrictEqual({
+        data: ["members-site", "real-dev", "real-dev-squad"],
```
Comment (Neha) [inline src/__tests__/useAxios.ts:20]: Move the data to Constant

File: src/__tests__/useAxios.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      })
+    );
+    await waitFor(() => expect(result.current.error).toBe(null));
+  });
+
+  it("should make an HTTP PATCH request and return the response data", async () => {
+    const patchUrl = "https://real-dev-squad.com/endpoint";
```
Comment (Neha) [inline src/__tests__/useAxios.ts:27]: Same as above...move to the constant

File: src/hooks/useAxios.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import axios, { AxiosRequestConfig } from "axios";
+import { useEffect, useState, useRef } from "react";
+
+type Methods = "get" | "post" | "put" | "delete" | "patch";
+
+type UseAxiosResult = {
+  data: any;
```
Comment (Neha) [inline src/hooks/useAxios.ts:7]: we should not be using 'any'. Try to use unknown.

File: src/hooks/useAxios.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+): UseAxiosResult => {
+  const intialState: UseAxiosResult = {
+    data: null,
+    loading: true,
+    error: null,
+  };
+
+  const cancelRequest = useRef<boolean>(false);
+
+  const [state, setState] = useState(intialState);
```
Comment (Neha) [inline src/hooks/useAxios.ts:25]: `useState< UseAxiosResult >`

File: src/hooks/useAxios.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  useEffect(() => {
+    cancelRequest.current = false;
```
Comment (Neha) [inline src/hooks/useAxios.ts:28]: what is current here mean?  If this is about status then rename this to status

**Review by Neha — COMMENTED** [review]
1. In test-cases, move all the hardcoded content to constant. 
2. Suggested a few rename and type suggestion.

**Pull request closed without merging**